### DISCUSSION TITLE
Add user to organization via REST API

### DISCUSSION THREAD

**Discussion by JonasDev17** [keycloak/keycloak#34230] [Q&A]
How do I add a user to an org? I looked at the REST API documentation but couldn't figure it out.
However I found this Issue which suggests that this is an existing feature: https://github.com/keycloak/keycloak/issues/27934
I use an autogenerated OpenAPI client.

**Comment by sneakylenny** [comment]
The documentation is unclear and my attempts at guessing what I have to do have been unsuccessful.

The documentation I follow: https://www.keycloak.org/docs-api/latest/rest-api/#_post_adminrealmsrealmorganizationsidmembers

It states: "Adds the user with the specified id as a member of the organization". However, it does not explain where the userId goes. It tells you the payload should be:

Name | Description | Default | Pattern
-- | -- | -- | --
body <br />_optional_ | [string] |   |  

I have tried to post the following

```php
$payload = [
    $testUserId
]
```

```php
$payload = [
    [$testUserId]
]
```

```php
$payload = [
    'body' => [$testUserId]
]
```

Using

```php
$client->post("/admin/realms/$realm/organizations/$organizationId/members", $payload);
```

Which all result in the same error:

```php
[
  "errorMessage" => "User does not exist"
]
```

Can this discussion please be re-opened as this issue has not been solved. If not I will create a new issue.

**Reply by JonasDev17** [reply]
You have to send the user id as a raw string or surrounded with double quotes ("json-ified") in the body of the request and the content type has to be application/json. Make sure the keycloak version is 26.06 or newer.

The original issue: https://github.com/keycloak/keycloak/issues/34401

**Reply by sneakylenny** [reply]
I have updated Keycloak to 26.0.7 along with sending the body like this:

```php
$client->post("/admin/realms/$realm/organizations/$organizationId/members", $testUserId);
```

And now it's working as intended.

Thanks for the quick response and the helpful reply!
